iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/AbstractTerminal.java

```diff
@@ -172,12 +177,12 @@ public boolean connect(Predicate<Switch> isTypeSwitchToOperate) {
         if (removed) {
             throw new PowsyblException(UNMODIFIABLE_REMOVED_EQUIPMENT + connectable.id);
         }
+        int variantIndex = getVariantManagerHolder().getVariantIndex();
+        String variantId = getVariantManagerHolder().getVariantManager().getVariantId(variantIndex);
+        boolean connectedBefore = isConnected();
+        connectable.notifyUpdate("beginConnect", variantId, connectedBefore, null);
         boolean connected = voltageLevel.connect(this, isTypeSwitchToOperate);
-        if (connected) {
-            int variantIndex = getVariantManagerHolder().getVariantIndex();
-            String variantId = getVariantManagerHolder().getVariantManager().getVariantId(variantIndex);
-            connectable.notifyUpdate("connection", variantId, false, true);
-        }
+        connectable.notifyUpdate("endConnect", variantId, !connected, connected);
```

Comment:
When the terminal was already connected, the `voltageLevel.connect(...)` method returns `false`.
Thus, the "endConnect" notification has the following values:
- `oldValue`: `true`;
- `newValue`: `false`.

Since `connect(...)` throws an `Exception` when a problem occurs, we may always use `true` for `newValue`.
It isn't a real update when the terminal is already connected, but I think it may be more important to have a notification when the process is finished:
```suggestion
        connectable.notifyUpdate("endConnect", variantId, connectedBefore, true);
```
<br/>

Another solution would be to use `null` as `newValue` when `connected` is `false`:
```suggestion
        connectable.notifyUpdate("endConnect", variantId, connectedBefore, connected ? true : null);
```

(To improve readability, I think it is better to use `connectedBefore` as `oldValue` parameter.)

Reply:
Ok, you are right. I was looking  at the Bus/Breaker implementation and not at the Node/Breaker one (which I should have done). It returns `false` if nothing was performed.
Maybe we can use the following values, then:
- `oldValue`: `connectedBefore`;
- `newValue`: `connectedBefore || connected` (true if it was already connected or if the connection succeeded)

Reply:
Ok, I just saw your change. This new policy is ok for me.

Reply:
I used the oldValue for the beginConnection and newValue for the endConnection and null for the other mainly because there is a filter of notif when oldValue = newValue. So I'm not really happy of that but maybe this is enough at the moment?